src/dynarec/rv64/dynarec_rv64_0f.c

```diff
@@ -2182,12 +2314,93 @@ uintptr_t dynarec64_0F(dynarec_rv64_t* dyn, uintptr_t addr, uintptr_t ip, int ni
             GETEM(x2, 0);
             MMX_LOOP_WS(x3, x4, MULW(x3, x3, x4));
             break;
+        case 0xD7:
```

Comment:
This looks so complicated, personally I would prefer simpler implementation for the scalar version.